src/H5FDmpio.c

```diff
@@ -964,6 +965,10 @@ H5FD__mpio_open(const char *name, unsigned flags, hid_t fapl_id, haddr_t H5_ATTR
     file->mpi_rank = mpi_rank;
     file->mpi_size = mpi_size;
 
+    /* Retrieve the flag indicating whether MPI_File_sync is needed after each write */
+    if (H5_mpio_get_file_sync_required(fh, &file->mpi_file_sync_required) < 0)
+        HDONE_ERROR(H5E_VFL, H5E_CANTGET, NULL, "unable to get mpi_file_sync_required hint")
```

Comment:
You may want this to be a goto_error instead of a done_error, since a failure to get the hint probably means we don't want to continue opening the file anyway.